commown_shareholder_register/tests/test_shareholder_register.py

```diff
@@ -0,0 +1,130 @@
+            {
+                "name": "Test Account Move",
+                "journal_id": cls.journal_1.id,
+                "date": date(*date_tuple),
+            }
+        )
+        if amount < 0:
+            attr2, attr1 = "credit", "debit"
+        else:
+            attr1, attr2 = "credit", "debit"
+
+        cls.account_move_lines |= cls.env["account.move.line"].create(
+            [
+                {
+                    "move_id": move.id,
+                    "account_id": account.id,
+                    "partner_id": partner.id,
+                    "date": date(*date_tuple),
+                    attr1: abs(amount),
+                },
+                {
+                    "move_id": move.id,
+                    "account_id": cls.account_balancing.id,
+                    "partner_id": partner.id,
+                    "date": date(*date_tuple),
+                    attr2: abs(amount),
+                },
+            ]
+        )
+
+    @classmethod
+    def setUpClass(cls):
+
+        super(TestShareholderRegister, cls).setUpClass()
+        cls.partner_1 = cls.env["res.partner"].create(
+            {
+                "name": "Partner 1",
+                "active": True,
+            }
+        )
+        cls.partner_2 = cls.env["res.partner"].create(
+            {
+                "name": "Partner 2",
+                "active": True,
+            }
+        )
+        cls.account_type_1 = cls.env["account.account.type"].create(
+            {"name": "Test Account Type 1", "type": "other"}
+        )
+        cls.account_porteur = cls.env["account.account"].create(
+            {
+                "name": "Test Account Porteur de Projet",
+                "code": "10134000",
+                "user_type_id": cls.account_type_1.id,
+            }
+        )
+        cls.account_soutient = cls.env["account.account"].create(
+            {
+                "name": "Test Account Soutient",
+                "code": "10136000",
+                "user_type_id": cls.account_type_1.id,
+            }
+        )
+        cls.account_beneficiaire = cls.env["account.account"].create(
+            {
+                "name": "Test Account Bénéficiaire",
+                "code": "10135000",
+                "user_type_id": cls.account_type_1.id,
+            }
+        )
+        cls.account_balancing = cls.env["account.account"].create(
+            {
+                "name": "Test Account For banancing journal",
+                "code": "XXXXXXXXXXX",
+                "user_type_id": cls.account_type_1.id,
+            }
+        )
+        cls.sequence_1 = cls.env["ir.sequence"].create(
+            {
+                "name": "Test Sequence 1",
+            }
+        )
+        cls.journal_1 = cls.env["account.journal"].create(
+            {
+                "name": "Test Journal 1",
+                "type": "bank",
+                "sequence_id": cls.sequence_1.id,
+            }
+        )
```

Comment:
On peut sûrement utiliser un journal existant qui convient, et donc économiser ce code ; pour le trouver facilement, utiliser ipdb et un code du genre `[j.get_xml_id() for j in cls.env["account.journal"].search([])]`

Reply:
Tu n'as pas utilisé ça ci-dessus, si pas clair on s'en parle.

Reply:
Les journaux n'ont pas d'id on a remplacé par une recherche sur le type
journal = cls.env["account.journal"].search([("type", "=", "general")], limit=1).ensure_one()